pkg/migration/generate.go

```diff
@@ -516,6 +636,34 @@ func (h *Handler) addMonitoringSpec(cluster *corev1.StorageCluster) error {
 		prometheus.Spec.ServiceMonitorSelector != nil &&
 		prometheus.Spec.ServiceMonitorSelector.MatchLabels["name"] == serviceMonitorName {
 		cluster.Spec.Monitoring.Prometheus.Enabled = true
+		cluster.Status.DesiredImages.Prometheus = *prometheus.Spec.Image
+
+		dep := &appsv1.Deployment{}
+		err := h.client.Get(
+			context.TODO(),
+			types.NamespacedName{
+				Name:      component.PrometheusOperatorDeploymentName,
```

Comment:
This should be `prometheusOpDeploymentName` as we should read the old prometheus deployment created by the spec generator.